src/System.Memory/src/System/MemoryExtensions.Portable.cs

```diff
@@ -57,10 +73,60 @@ public static int CompareTo(this ReadOnlySpan<char> span, ReadOnlySpan<char> val
             {
                 return span.SequenceCompareTo(value);
             }
+            else if (comparisonType == StringComparison.OrdinalIgnoreCase)
+            {
+                return CompareToOrdinalIgnoreCase(span, value);
+            }
 
             return string.Compare(span.ToString(), value.ToString(), comparisonType);
         }
 
+        // Borrowed from https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Globalization/CompareInfo.cs#L539
```

Comment:
>Borrowed from https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Globalization/CompareInfo.cs#L539 [](start = 11, length = 120)

as the code in shared folder, would it be possible to factor the code and have only one method that can be called for the string and span? so, we'll have one place to maintain?

Reply:
It wouldn't be a trivial change since we would have to add ifdefs within the existing CompareToOrdinalIgnoreCase implementation and factor that out into a separate file. We can't just include the whole file for netfx/netstandard (note, this optimization is not for .net core).